baybe/searchspace/discrete.py

```diff
@@ -382,3 +390,7 @@ def parameter_cartesian_prod_to_df(
     ret = pd.DataFrame(index=index).reset_index()
 
     return ret
+
+
+# Register deserialization hook
+converter.register_structure_hook(SubspaceDiscrete, select_constructor_hook)
```

Comment:
so this constructor hook seems rather general, is there any reason why its only enabled for the searchspace and not immediately for every object?

Reply:
Upfront: have you seen my note in the PR description? I think this not yet the final version but will also depend on how it's going to be included into cattrs. But to answer the question: simply because it would be inefficient. Following that same logic, we could also have a general "if object == my_type" check at the beginning of all hooks we write and then apply it to ALL python objects instead of just to "my_type" – you know what I mean? The hook would be called for no reason. I think the clean way would be to either manually register it to exactly the types we need (current approach) or perhaps use some predicate function to decide, but I wouldn't bother until the question what happens on cattrs side is fully answered. Will draft a cattrs PR for it and then we'll see. Please reactivate if you want to further discuss.